src/geocel/GeantGeoParams.cc

```diff
@@ -383,6 +383,56 @@ std::vector<inp::Surface> make_inp_surfaces(GeantGeoParams const& geo)
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Create sensitive detectors input from Geant4 sensitive detectors.
+ */
+std::vector<inp::Detector> make_inp_detectors(GeantGeoParams const& geo)
+{
+    // Process detectors
+    std::vector<inp::Detector> result;
+
+    auto const& vol_labels = geo.impl_volumes();
+
+    std::unordered_map<G4VSensitiveDetector const*, std::vector<VolumeId>>
+        detector_map;
+
+    // Process each logical volume
+    for (auto iv_id : range(ImplVolumeId{vol_labels.size()}))
+    {
+        auto vol_id = geo.volume_id(iv_id);
+        if (!vol_id)
+        {
+            // This volume isn't part of the world hierarchy
+            continue;
+        }
+        auto& g4lv = *geo.id_to_geant(vol_id);
+
+        G4VSensitiveDetector const* sd = g4lv.GetSensitiveDetector();
+        // Add volume id to detector map if it is in a detector region
+        if (sd)
```

Comment:
One idiom we often use is to limit the scope of a variable by declaring it inside the `if` condition:
```suggestion
        // Add volume id to detector map if it is in a detector region
        if (auto const* sd = g4lv.GetSensitiveDetector())
```